docs/scenarios_authoring/Spel.md

```diff
@@ -264,18 +264,96 @@ can be accessed in e.g. in following ways:
 * `#exampleObjects['someNestedObject']['someFieldInNestedObject']`
 * `#exampleObjects['someArrayWithObjects'][0]['someFieldInObjectInArray']`
 
-Every unknown accessed field/element will produce `Unknown` data type which can be further navigated or [cast](#Casting) to a desired type.
+Every unknown accessed field/element will produce `Unknown` data type, which can be further navigated or [converted](#type-conversions) to a desired type.
 
 ### Type conversions
 
-It is possible to convert from a type to another type and this can be done by implicit and explicit conversion.
+It is possible to convert or cast from a type to another type and this can be done by implicit and explicit conversion.
+
+#### Explicit conversions
+
+Explicit conversions/casts are available as built-in functions.
+List of built-in functions:
+- `canBe(className)`/`to(className)`/`toOrNull(className)`
+- `canBeBoolean`/`toBoolean`/`toBooleanOrNull`
+- `canBeLong`/`toLong`/`toLongOrNull`
+- `canBeDouble`/`toDouble`/`toDoubleOrNull`
+- `canBeBigDecimal`/`toBigDecimal`/`toBigDecimalOrNull`
+- `canBeList`/`toList`/`toListOrNull`
+- `canBeMap`/`toMap`/`toMapOrNull`
+
+The aforementioned functions first attempt to cast a value to the specified class. If the cast fails and there is a 
+defined conversion to that class, the conversion is applied.
+The `canBe`, `to` and `toOrNull` functions take the name of target class as a parameter, in contrast to, for
+example, `canBeLong` which has the name of target class in the function name. 
+
+Functions with the prefix `canBe` check whether a type can be cast or converted to the appropriate type. Functions with
+the `to` prefix cast or convert a value to the desired type, and if the operation fails, an exception is propagated
+further. Functions with the `to` prefix and `OrNull` suffix cast or convert a value to the desired type,
+and if the operation fails, a null value is returned.
+
+Examples of conversions:
+
+| Expression                 | Result | Type    |
+|----------------------------|--------|---------|
+| `"123".canBeDouble`        | true   | Boolean |
+| `"123".toDouble`           | 123.0  | Double  |
+| `"abc".toDoubleOrNull`     | null   | Double  |
+| `"123".canBe('Double')`    | true   | Boolean |
+| `"123".to('Double')`       | 123.0  | Double  |
+| `"abc".toOrNull('Double')` | null   | Double  |
```

Comment:
Let's add an example that is failing in runtime

Reply:
Also, let's add an example of conversion of some list to map using projection mechanism as it is tricky. For example:
```
{12, 23, 34}.![{key: 'key_' + #this, value: #this}]
```